pkg/proxy/http.go

```diff
@@ -72,7 +53,33 @@ func (p *Proxy) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 			}
 		}
 	}
-	log.Infof("received request, reqHost: %s", reqHost)
+	log.Infof("proxy received request, reqHost: %s", reqHost)
+
+	if values := req.Header[HeaderOrigDest]; len(values) > 0 {
+		origDest = values[0]
+		req.Header.Del(HeaderOrigDest)
+
+		if idx := strings.LastIndex(origDest, ":"); idx >= 0 {
+			origDestIp = origDest[:idx]
+			if v, err := strconv.Atoi(origDest[idx+1:]); err != nil {
+				http.Error(w, fmt.Sprintf("invalid header %s value: %s", HeaderOrigDest, origDest), http.StatusBadRequest)
+				return
+			} else {
+				origDestPort = v
+			}
+		} else {
+			origDestIp = origDest
+		}
+	}
+
+	if origDest == "" {
+		if idx := strings.LastIndex(reqHost, ":"); idx >= 0 {
+			origDestIp = reqHost[:idx]
+		} else {
+			origDestIp = reqHost
```

Comment:
have we confirmed that the `Dial` later support `host` format addr?

Reply:
host can be ip or fqdn